source/agora/node/Fee.d

```diff
@@ -155,3 +160,95 @@ unittest
     assert(calculateDataFee(490, 200) == Amount(105_900_000L));
     assert(calculateDataFee(500, 200) == Amount(111_800_000L));
 }
+
+/*******************************************************************************
+
+    A class that provides the ability to verify `DataPayload` stored
+    in a `Transaction`.
+
+*******************************************************************************/
+
+public class DataPayloadChecker
+{
+    /// The address of commons budget
+    public PublicKey commons_budget_address;
+
+    /// The maximum size of data payload
+    public uint max_size_payload;
+
+    /// The factor to calculate for the fee of data payload
+    public uint payload_fee_factor;
+
+    /// Ctor
+    public this (in PublicKey commons_budget_address,
+                 uint max_size_payload,
+                 uint payload_fee_factor)
+    {
+        this.commons_budget_address = commons_budget_address,
+        this.max_size_payload = max_size_payload;
+        this.payload_fee_factor = payload_fee_factor;
+    }
+
+    /***************************************************************************
+
+        Checks the following in the data payload:
+
+            Checks that the size of the data does not exceed
+            the maximum size allowed.
+            Checks if a commons budget address exists in the output.
+            Checks if the appropriate fee is paid by commons budget address.
+
+        Params:
+            tx = `Transaction`
+            sum_unspent = Sum of unspent amounts
+
+        Return:
+            `null` if the transaction is valid, a string explaining the reason it
+            is invalid otherwise.
+
+    ***************************************************************************/
+
+    public string check (in Transaction tx, in Amount sum_unspent) nothrow @safe
```

Comment:
https://github.com/bpfkorea/agora/pull/1321#issuecomment-727713270
The above link was proposed by @AndrejMitrovic  in the previous PR.
I tried to make a simple delegate that only checks fees according to @AndrejMitrovic 's suggestion.
But I did this because I needed more information (Comments Budget Address, ...).

If this is to be changed to a simple delegate to check fee, then ConsensusParam must be delivered to `agora/consensus/validation/Transaction.d`
I did this to reduce dependencies, but please let me know if you have any other opinions.

Reply:
Ah @MichaelKim20, please remove the `sum_unspent` parameter. I was mistaken that we need it. The fee is in the output for the commons budget, just like you set it here.